source/php/Module/views/recommend.blade.php

```diff
@@ -1,40 +1,44 @@
 @if (!$hideTitle)
     @typography([
-        'element' => "h2",
-        'classList' => ['u-margin--0']
+        'element' => 'h4',
+        'variant' => 'h2',
+        'classList' => ['module-title']
     ])
-        {{ $postTitle }}
+    {{ $postTitle }}
     @endtypography
 @endif
 
-@if($enableRekAI)
-    <div id="{{$recommendUid}}" class="mod-recommend__items">
-        @if($recommendLinkList) 
-            @foreach($recommendLinkList as $recommendLink)
-                @include('partials.button', [
-                    "text" => $recommendLink->recommendLinkLabel,
-                    "href" => $recommendLink->recommendLinkTarget,
-                    "type" => "static",
-                ])
-            @endforeach
-        @endif
+@include($template ? $template . '-template' : 'button-template')
 
-        <!-- Preloader -->
-        @include('partials.preload', [
-            'rekaiNumberOfRecommendation' => $rekaiNumberOfRecommendation
-        ])
-    </div>
+@if($enableRekAI)
     <script>
-        window.addEventListener("rekai.load", function(){
+        window.addEventListener("rekai.load", function() {
+          function createTemplateItem(data, template) {
+            let view = '';
+            switch(template) {
+              case "card":
+                view = '<?php echo modularity_recommend_render_blade_view("partials.card", ["heading" => "{MOD_RECOMMEND_TITLE}", "content" => "{MOD_RECOMMEND_CONTENT}", "href" => "{MOD_RECOMMEND_HREF}"]); ?>';
+                break;
+              case "button":
+                view = '<?php echo modularity_recommend_render_blade_view("partials.button", ["href"=> "{MOD_RECOMMEND_HREF}", "text" => "{MOD_RECOMMEND_TITLE}", "type" => "dynamic"]); ?> ';
+                break;
+              default:
+            }
+
+            view = view.replace("{MOD_RECOMMEND_HREF}", data.url ?? '');
+            view = view.replace("{MOD_RECOMMEND_TITLE}", data.title ?? '');
+            view = view.replace("{MOD_RECOMMEND_CONTENT}", data.ingress ?? '');
+
+            return view;
+          }
+
             function renderHtml(data) {
-                
                 let rekAiInputString = '';
                 let targetId = document.getElementById("{{$recommendUid}}");
-                
-                if(targetId) {
 
+                if(targetId) {
                     //Remove the preloader
-                    let preloaderItems = targetId.querySelectorAll(".u-preloader");
+                    let preloaderItems = targetId.querySelectorAll(".u-preloader, .o-grid");
```

Comment:
Remove specific targets, add class to elements. eg. "rek-ai-preload-remove" to target this to be removed.